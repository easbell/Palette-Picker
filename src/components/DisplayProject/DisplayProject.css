.project-card {
  width: 40%;
  margin: 30px 10px 0px 10px;
  align-items: space-between;
  border-radius: 10px;
  background: #f4f4f4;
  color: #444;
  text-align: left;
  padding: 20px;
  box-shadow: 0 3px 5px rgba(0,0,0,0.20);
}

.project-card,
.edit-project-card {
  max-width: 500px;
  min-width: 440px;
}

.edit-project-card {
  width: 80%;
  margin: auto;
  margin-top: 30px;
}

.project-name {
  color: #444;
  background: #f4f4f4;
  width: 100%;
  border: none;
  border-bottom: solid #ddd 2px;
  padding: 0px 0px 10px;
  font-size: 1.5rem;
  margin-bottom: 30px;
}

.project-name:focus {
  outline: none;
}

.delete-btn-container {
  display: flex;
  justify-content: flex-end;
  padding: 30px 0px 0px;
}

.update-project {
  background: #ea4c88;
  color: #fff;
  padding: 10px 0px;
  font-size: 13px;
  font-weight: bold;
  letter-spacing: .02em;
  line-height: 17px;
  vertical-align: middle;
  cursor: pointer;
  border: none;
  text-decoration: none;
  border-radius: 4px;
  margin-left: 10px;
  width: 120px;  
}

.update-project:hover {
  background-color: #ea4c88ad;
  transition: all .2s ease;
}