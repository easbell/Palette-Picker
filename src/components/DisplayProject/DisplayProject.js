import React from 'react'
import { connect } from 'react-redux'
import './DisplayProject.css'
import DisplayPalette from '../DisplayPalette/DisplayPalette'
import { Link } from 'react-router-dom'

export const DisplayProject = (props) => {
    const { name, id } = props.project
    const palettes = props.palettes.filter(palette => {
      return palette.project_id === id
    }).map(palette => {
      return <DisplayPalette key={palette.id} palette={palette}/>
    })

    return(
      <div className='project-card'>
        <Link to={`/my-projects/${id}`} style={{ textDecoration: 'none' }}>
          <h2 className='project-name'>
            {name}
          </h2>
          {palettes}
        </Link>
      </div>
    )
}

const mapStateToProps = (state) => ({
  palettes: state.palettes
})

export default connect(mapStateToProps)(DisplayProject)
