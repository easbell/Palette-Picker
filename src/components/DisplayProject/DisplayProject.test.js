import React from 'react'
import { DisplayProject, mapStateToProps } from './DisplayProject'
import { shallow } from 'enzyme'

describe('DisplayProject', () => {
  let wrapper
  let mockProps = {
    palettes: [{ id: 1, project_id: 1 }, { id: 2, project_id: 3 }],
    project: { id: 1, name: 'my project'}
  }

  it('should match the snapshot', () => {
    wrapper = shallow(<DisplayProject {...mockProps} />)
    expect(wrapper).toMatchSnapshot()
  })

  describe('mapStateToProps', () => {
    it('returns correct props', () => {
      const mockState = {
        palettes: ['palettes']
      }
      const expectedProps = {
        palettes: ['palettes']
      }

      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expectedProps)
    });
  });
})