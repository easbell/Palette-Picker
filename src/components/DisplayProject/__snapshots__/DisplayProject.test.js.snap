// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayProject should match the snapshot 1`] = `
<div
  className="project-card"
>
  <Link
    style={
      Object {
        "textDecoration": "none",
      }
    }
    to="/my-projects/1"
  >
    <h2
      className="project-name"
    >
      my project
    </h2>
    <DisplayPalette
      key="1"
      palette={
        Object {
          "id": 1,
          "project_id": 1,
        }
      }
    />
  </Link>
</div>
`;
